--- a/library/cpp/token/token_structure.h
+++ b/library/cpp/token/token_structure.h
@@ -134,20 +134,20 @@ public:
         Tokens.pop_back();
     }
     const TCharSpan* begin() const {
-        return Tokens.begin();
+        return Tokens.data();
     }
     const TCharSpan* end() const {
-        return Tokens.end();
+        return Tokens.data() + Tokens.size();
     }
     TCharSpan* begin() {
-        return Tokens.begin();
+        return Tokens.data();
     }
     TCharSpan* end() {
-        return Tokens.end();
+        return Tokens.data() + Tokens.size();
     }
     const TCharSpan* cbegin() const {
-        return Tokens.cbegin();
+        return Tokens.data();
     }
     const TCharSpan* cend() const {
-        return Tokens.cend();
+        return Tokens.data() + Tokens.size();
     }
