--- a/library/cpp/hnsw/index_builder/index_writer.cpp
+++ b/library/cpp/hnsw/index_builder/index_writer.cpp
@@ -20,17 +20,18 @@
         TVector<const ui32*> levels;
         TVector<ui32> numNeighborsInLevels;
         TVector<ui32> numItemsInLevels;
+        const ui32* flatLevelsEnd = index.FlatLevels.data() + index.FlatLevels.size();
         {
-            const ui32* data = index.FlatLevels.begin();
+            const ui32* data = index.FlatLevels.data();
             for (i64 numItems = index.NumItems; numItems > 1; numItems /= index.LevelSizeDecay) {
-                Y_ENSURE(data < index.FlatLevels.end());
+                Y_ENSURE(data < flatLevelsEnd);
                 levels.push_back(data);
                 numNeighborsInLevels.push_back(Min<i64>(index.MaxNeighbors, numItems - 1));
                 numItemsInLevels.push_back(numItems);
                 data += numItems * numNeighborsInLevels.back();
             }
 
-            Y_ENSURE(data == index.FlatLevels.end());
+            Y_ENSURE(data == flatLevelsEnd);
         }
 
         for (auto levelNum : xrange<i64>(levels.size() - 1, -1, -1)) {
