--- a/catboost/python-package/setup.py
+++ b/catboost/python-package/setup.py
@@ -98,7 +98,7 @@ def get_setup_requires(argv):
     setup_requires = ['cmake', 'wheel', 'conan ~= 2.4.1', 'cython ~= 3.0.10', 'numpy']

     if ('build_widget' in argv) or (not ('--no-widget' in argv)):
-        setup_requires += ['jupyterlab (>=3.0.6, <3.6.0)']
+        setup_requires += ['jupyterlab (>=3.0.6)']
     return setup_requires


