--- a/contrib/libs/tcmalloc/no_percpu_cache/CMakeLists.txt
+++ b/contrib/libs/tcmalloc/no_percpu_cache/CMakeLists.txt
@@ -34,6 +34,9 @@
 add_global_library_for(libs-tcmalloc-no_percpu_cache.global
   libs-tcmalloc-no_percpu_cache)

+# Force C++ linker for tcmalloc since it contains C++ code with C++ runtime dependencies
+set_target_properties(libs-tcmalloc-no_percpu_cache.global PROPERTIES LINKER_LANGUAGE CXX)
+
 target_compile_options(libs-tcmalloc-no_percpu_cache.global PRIVATE
   -DTCMALLOC_256K_PAGES
   -DTCMALLOC_DEPRECATED_PERTHREAD
