--- a/catboost/libs/data/data_provider_builders.cpp
+++ b/catboost/libs/data/data_provider_builders.cpp
@@ -1661,8 +1661,8 @@
             } else {
                 size_t prefixLength = Min(Data.MetaInfo.ClassLabels.size(), schemaClassLabels.size());
                 auto firstGivenLabels = TConstArrayRef<NJson::TJsonValue>(
-                    Data.MetaInfo.ClassLabels.begin(),
-                    Data.MetaInfo.ClassLabels.begin() + prefixLength
+                    Data.MetaInfo.ClassLabels.data(),
+                    Data.MetaInfo.ClassLabels.data() + prefixLength
                 );
                 CB_ENSURE(firstGivenLabels == schemaClassLabels,
                           "Class-names incompatible with quantized pool, expected: " << JoinSeq(",", schemaClassLabels));
