--- a/catboost/libs/data/quantization.cpp
+++ b/catboost/libs/data/quantization.cpp
@@ -62,8 +62,8 @@
             dstIndices.yresize(srcSubsetIndexing.Size());
             Iota(dstIndices.begin(), dstIndices.end(), ui32(0));

-            TDoubleArrayIterator<ui32, ui32> beginIter{srcIndices.begin(), dstIndices.begin()};
-            TDoubleArrayIterator<ui32, ui32> endIter{srcIndices.end(), dstIndices.end()};
+            TDoubleArrayIterator<ui32, ui32> beginIter{srcIndices.data(), dstIndices.data()};
+            TDoubleArrayIterator<ui32, ui32> endIter{srcIndices.data() + srcIndices.size(), dstIndices.data() + dstIndices.size()};

             Sort(beginIter, endIter, [](auto lhs, auto rhs) { return lhs.first < rhs.first; });
