--- a/catboost/libs/helpers/polymorphic_type_containers.h
+++ b/catboost/libs/helpers/polymorphic_type_containers.h
@@ -290,11 +290,11 @@
             std::visit(
                 [&] (const auto& subsetIndexingAlternative) {
                     CreateSubRangesIterators(
                         subsetIndexingAlternative,
                         definedApproximateBlockSize,
                         subRangeIterators,
                         subRangeStarts
                     );
                 },
-                *SubsetIndexing
+                static_cast<const typename TArraySubsetIndexing<ui32>::TBase&>(*SubsetIndexing)
             );
         }
